cmd/node/node.go

```diff
@@ -458,6 +459,14 @@ func (app *App) SetLogLevel(name, loglevel string) error {
 	return nil
 }
 
+type nonceFetcherAdapter struct {
+	cdb *datastore.CachedDB
+}
+
+func (nfa nonceFetcherAdapter) VRFNonce(nodeID types.NodeID, epoch types.EpochID) (types.VRFPostIndex, error) {
+	return atxs.VRFNonce(nfa.cdb, nodeID, epoch)
+}
```

Comment:
i think i will prefer creating a option for each component's `NewXXXX` function.
that option will default to a function that executes `txs.VRFNonce(cdb, nodeID, epoch)` (because cdb is already passed to most components), and gets overwritten in tests

Reply:
similar to this https://github.com/spacemeshos/go-spacemesh/blob/3c2f1837675840e69e3f5bb5cabbd9a9e54807ce/proposals/handler.go#L79

Reply:
I added a few functional options to inject a mocked `nonceFetcher` in tests. If not used a standard implementation is used that reads the nonce from the ATX table using `sql/atxs`